Libraries/Image/resolveAssetSource.js

```diff
@@ -104,9 +110,13 @@ function getBasePath(asset) {
  */
 function getScaledAssetPath(asset) {
```

Comment:
@zahanm, for absolute clarity, look at the unit tests.

On iOS,
Bundle Location: 'file:///Path/To/Sample.app/main.bundle' 
httpServerLocation: '/assets/module/a/'
Name: 'logo'
type: 'png'
**Resolved Asset location: '/Path/To/Sample.app/assets/module/a/logo.png'**

On Android,
Bundle Location: '/sdcard/Path/To/main.bundle'
httpServerLocation: '/assets/module/a/',
name: 'logo'
type: 'png'
**Resolved Asset location: '/sdcard/Path/To/drawable-mdpi/module_a_logo.png'**